AuthenticatorPro.Shared/Source/Data/Generator/SteamOtp.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Text;
+using OtpNet;
+
+namespace AuthenticatorPro.Shared.Data.Generator
+{
+    public class SteamOtp : IGenerator
+    {
+        public const string Alphabet = "23456789BCDFGHJKMNPQRTVWXY";
+        public const int Digits = 5;
+
+        private readonly OtpNet.Totp _totp;
+
+        public GenerationMethod GenerationMethod => GenerationMethod.Time;
+
+        public SteamOtp(string secret, int digits)
+        {
+            var secretBytes = Base32Encoding.ToBytes(secret);
+            _totp = new SteamTotp(secretBytes, digits);
+        }
+
+        public string Compute()
+        {
+            return _totp.ComputeTotp();
+        }
+
+        public DateTime GetRenewTime() {
+            return DateTime.UtcNow.AddSeconds(_totp.RemainingSeconds());
+        }
+
+        private class SteamTotp : OtpNet.Totp
+        {
+            private readonly int _digits;
+
+            public SteamTotp(byte[] secretKey, int digits = 5) : base(secretKey, 30, OtpHashMode.Sha1, digits)
```

Comment:
Since Steam always generates 5 digit codes, perhaps there is no need to have a parameter here. Instead move the outer const int Digits to SteamTotp and rename to NumDigits so it doesn't conflict with the Digits method. Pass the constant directly into the base call.